internal.lua

```diff
@@ -63,28 +63,57 @@ function digiline:rules_link_anydir(output, input)
 	or     digiline:rules_link(input, output)
 end
 
-function digiline:transmit(pos, channel, msg, checked)
-	local checkedid = tostring(pos.x).."_"..tostring(pos.y).."_"..tostring(pos.z)
-	if checked[checkedid] then return end
-	checked[checkedid] = true
+local function queue_new()
+	return {nextRead = 1, nextWrite = 1}
+end
 
-	local node = minetest.get_node(pos)
-	local spec = digiline:getspec(node)
-	if not spec then return end
+local function queue_empty(queue)
+	return queue.nextRead == queue.nextWrite
+end
 
+local function queue_push(queue, object)
+	local nextWrite = queue.nextWrite
+	queue[nextWrite] = object
+	queue.nextWrite = nextWrite + 1
+end
 
-	-- Effector actions --> Receive
-	if spec.effector then
-		spec.effector.action(pos, node, channel, msg)
-	end
+local function queue_pop(queue)
+	local nextRead = queue.nextRead
+	local object = queue[nextRead]
+	queue[nextRead] = nil
+	queue.nextRead = nextRead + 1
+	return object
+end
 
-	-- Cable actions --> Transmit
-	if spec.wire then
-		local rules = digiline:importrules(spec.wire.rules, node)
-		for _,rule in ipairs(rules) do
-			if digiline:rules_link(pos, digiline:addPosRule(pos, rule)) then
-				digiline:transmit(digiline:addPosRule(pos, rule), channel, msg, checked)
+function digiline:transmit(pos, channel, msg, checked)
+	digiline:vm_begin()
+	local queue = queue_new()
+	queue_push(queue, pos)
+	while not queue_empty(queue) do
+		local curPos = queue_pop(queue)
+		local node = digiline:get_node_force(curPos)
+		local spec = digiline:getspec(node)
+		if spec then
+			-- Effector actions --> Receive
+			if spec.effector then
+				spec.effector.action(curPos, node, channel, msg)
+			end
+
+			-- Cable actions --> Transmit
+			if spec.wire then
+				local rules = digiline:importrules(spec.wire.rules, node)
+				for _, rule in ipairs(rules) do
+					local nextPos = digiline:addPosRule(curPos, rule)
+					if digiline:rules_link(curPos, nextPos) then
+						local checkedID = tostring(nextPos.x) .. "_" .. tostring(nextPos.y) .. "_" .. tostring(nextPos.z)
```

Comment:
I know this is the original method, but why not hash_node_position? Likely faster than constructing a string with `..`s.